### PR TITLE
DOC-5751 add regions column for show backup

### PR DESCRIPTION
**Pull request by ashleykwtam**
Addresses DOC-5751:
-  Adds the 'regions' column to SHOW BACKUP which outputs a string of ALTER DATABASE commands if
the database is a multi-region database and NULL if not

Preview:
- [SHOW BACKUP](https://deploy-preview-17079--cockroachdb-docs.netlify.app/docs/dev/show-backup.html#response)

### CODE REVIEW

File: v23.1/show-backup.md [hunk 1]
```diff
@@ -77,6 +77,7 @@ Field | Description
 `rows` | Number of rows in tables that are part of the backup.
 `create_statement` | The `CREATE` statement used to create [table(s)](create-table.html), [view(s)](create-view.html), or [sequence(s)](create-sequence.html) that are stored within the backup. This displays when `SHOW BACKUP SCHEMAS` is used. Note that tables with references to [foreign keys](foreign-key.html) will only display foreign key constraints if the table to which the constraint relates to is also included in the backup.
 `is_full_cluster` |  Whether the backup is of a full cluster or not.
+`regions` | Whether the database is multi-region or not. If the database is a multi-region database, a string of [`ALTER DATABASE`](alter-database.html) commands will show. If the database is single-region, `NULL` will show.
```
Comment (msbutler) [inline v23.1/show-backup.md:80]: I'd rephrase this like the `create_statement` explanation above: "The `ALTER DATABASE` statements used to configure the database with multi region zone configurations, if they exist".

This link may be more appropriate: https://www.cockroachlabs.com/docs/dev/multiregion-overview.html#database-regions

File: v23.1/show-backup.md [hunk 2]
```diff
@@ -77,6 +77,7 @@ Field | Description
 `rows` | Number of rows in tables that are part of the backup.
 `create_statement` | The `CREATE` statement used to create [table(s)](create-table.html), [view(s)](create-view.html), or [sequence(s)](create-sequence.html) that are stored within the backup. This displays when `SHOW BACKUP SCHEMAS` is used. Note that tables with references to [foreign keys](foreign-key.html) will only display foreign key constraints if the table to which the constraint relates to is also included in the backup.
 `is_full_cluster` |  Whether the backup is of a full cluster or not.
+`regions` | The [`ALTER DATABASE`](alter-database.html) statement(s) used to configure the database with [multi-region zone configurations](multiregion-overview.html#database-regions), if they exist. If the database is single-region, `NULL` will show.
```
Comment (msbutler) [inline v23.1/show-backup.md:80]: s/single-region/apart of single region cluster configuration/r

In an MR cluster configuration, all Databases will need to be configured with an `ALTER DATABASE` cmd. See these [instructions](https://www.cockroachlabs.com/docs/v23.1/multiregion-overview#overview). In an MR cluster configuration, you could still have a database constrained to a single region.

**Review by msbutler — APPROVED** [review]

File: v23.1/show-backup.md [hunk 3]
```diff
@@ -115,18 +116,19 @@ SHOW BACKUP FROM LATEST IN 'external://backup_s3';
...
-system        | public             | jobs                       | table       | incremental | 2022-04-08 14:23:55.33557 | 2022-04-08 14:26:01.699694 |      30148 |    23 |      true
-system        | public             | locations                  | table       | incremental | 2022-04-08 14:23:55.33557 | 2022-04-08 14:26:01.699694 |        261 |     5 |      true
+database_name | parent_schema_name |        object_name         | object_type | backup_type | start_time |          end_time          | size_bytes | rows | is_full_cluster |                                             regions
+----------------+--------------------+----------------------------+-------------+-------------+------------+----------------------------+------------+------+-----------------+--------------------------------------------------------------------------------------------------
+NULL          | NULL               | movr                       | database    | full        | NULL       | 2023-05-23 15:27:33.251745 |       NULL | NULL |        f        | ALTER DATABASE movr SET PRIMARY REGION "us-east-2"; ALTER DATABASE movr ADD REGION "us-west-1";
```
Comment (MichaelTrestman) [inline v23.1/show-backup.md:121]: I just noticed that the value of `is_full_cluster` is 'f' here but is true/false elsewhere. It should be made consistent/updated (apologies if I'm missing something about the query that explains this).

**Review by MichaelTrestman — APPROVED** [review]
Unless I'm missing something, it looks like there's an inconsistency in the output, should this output be updated/validated/corrected throughout? Perhaps this is beyond the scope of this current ticket.
LGTM other than that.

**Comment by ashleykwtam** [comment]
Merging in a separate branch created https://github.com/cockroachdb/docs/pull/17243 as this was done before the `main` branch cut
